src/Features/Core/Portable/PullMemberUp/AbstractPullMemberUpRefactoringProvider.cs

```diff
@@ -0,0 +1,110 @@
+    {
+        protected abstract bool IsSelectionValid(TextSpan span, SyntaxNode selectedMemberNode);
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            // Currently support to pull field, method, event, property and indexer up,
+            // constructor, operator and finalizer are excluded.
+            var document = context.Document;
+            var semanticModel = await document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var selectedMemberNode = root.FindNode(context.Span);
+
+            if (selectedMemberNode == null)
+            {
+                return;
+            }
+
+            var selectedMember = semanticModel.GetDeclaredSymbol(selectedMemberNode);
+            if (selectedMember == null || selectedMember.ContainingType == null)
+            {
+                return;
+            }
+
+            if (!selectedMember.IsKind(SymbolKind.Property) &&
+                !selectedMember.IsKind(SymbolKind.Event) &&
+                !selectedMember.IsKind(SymbolKind.Field) &&
+                !selectedMember.IsKind(SymbolKind.Method))
+            {
+                // Static, abstract and accessiblity are not checked here but in PullMemberUpAnalyzer.cs since there are
+                // two refactoring options provided for pull members up,
+                // 1. Quick Action (Only allow members that don't cause error)
+                // 2. Dialog box (Allow modifers may cause errors and will provide fixing)
+                return;
+            }
+
+            if (selectedMember is IMethodSymbol methodSymbol && !methodSymbol.IsOrdinaryMethod())
+            {
+                return;
+            }
+
+            if (!IsSelectionValid(context.Span, selectedMemberNode))
+            {
+                return;
+            }
+
+            var allDestinations = FindAllValidDestinations(
+                selectedMember,
+                document.Project.Solution,
+                context.CancellationToken);
+            if (allDestinations.Length == 0)
+            {
+                return;
+            }
+            
+            PullMemberUpViaQuickAction(context, selectedMember, allDestinations);
+        }
+
+        private ImmutableArray<INamedTypeSymbol> FindAllValidDestinations(
+            ISymbol selectedMember,
+            Solution solution,
+            CancellationToken cancellationToken)
+        {
+            var containType = selectedMember.ContainingType;
+            var allDestinations = selectedMember.IsKind(SymbolKind.Field)
+                ? containType.GetBaseTypes().ToImmutableArray()
+                : containType.AllInterfaces.Concat(containType.GetBaseTypes()).ToImmutableArray();
+
+            return allDestinations.WhereAsArray(baseType => baseType != null &&
```

Comment:
fwiw, baseType shuld never be null here.  but it's not very problematic to be overly cautious.